https://redhat.atlassian.net/browse/ACM-31036 (#8682)
open-cluster-management/rhacm-docs: release_notes/known_issues_install.adoc

Follow the link:../install/cleanup_reinstall.adoc#cleanup-reinstall[Cleaning up artifacts before reinstalling] procedure to prevent this problem.

link:https://issues.redhat.com/browse/ACM-14343[ACM-14343]

[#add-on-stuck-upgrade]
== The _kubevirt-hyperconverged_ add-on is stuck in `in progress` status during upgrade from {acm-short} {product-version-prev} to {product-version}

> During an upgrade from {acm-short} {product-version-prev} to {product-version}, the `kubevirt-hyperconverged-operator` resource is not cleaned up and you receive the following error in the `ManifestWork`:

Reviewer: ```suggestion
During an upgrade from {acm-short} {product-version-prev} to {product-version}, the `kubevirt-hyperconverged-operator` resource is not cleaned up and you receive the following error in the `ManifestWork` resource:
```

[source,bash]
----
Failed to apply manifest: operatorpolicies.policy.open-cluster-management.io
"kubevirt-hyperconverged-operator" is forbidden: User
"system:serviceaccount:open-cluster-management-agent:klusterlet-work-sa"
cannot get resource "operatorpolicies" in API group
"policy.open-cluster-management.io" in the namespace "open-cluster-management-policies"
----

To resolve the issue, complete the following steps:

> 
> . Remove existing resources. Run the following commands:
> 
> +
> [source,bash]
> ----
> oc delete clustermanagementaddon kubevirt-hyperconverged-operator --ignore-not-found
> oc delete addontemplate kubevirt-hyperconverged-operator --ignore-not-found
> ----

Reviewer: Recommend adding a line break to separate the commnads

```suggestion
. Remove existing resources. Run the following commands:

+
[source,bash]
----
oc delete clustermanagementaddon kubevirt-hyperconverged-operator --ignore-not-found

oc delete addontemplate kubevirt-hyperconverged-operator --ignore-not-found
----
```

Author: @dockerymick I normally see these single spaced and it's fine, as they are ran at the same time. Is this a necessary change?

Author: created a space

Reviewer: It was not necessary , thanks for accepting !

+
[source,bash]
----
oc delete clustermanagementaddon kubevirt-hyperconverged
----

. Get the namespace that is specified in the `installer.namespace` label of the `kubevirt-hyperconverged` `ClusterManagementAddOn` resource. Run the following command and record the output to use in the next step:

+
[source,bash]
----
oc get clustermanagementaddon kubevirt-hyperconverged -o jsonpath='{.metadata.labels.installer\.namespace}'
----

. Create the `ClusterManagementAddOn` resource. See the following example and replace `<installer_namespace>` with the value that you retrieved in the previous step:

+
[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1beta1
kind: ClusterManagementAddOn
metadata:
  name: kubevirt-hyperconverged
  annotations:
    addon.open-cluster-management.io/lifecycle: "addon-manager"
spec:
  addOnMeta:
    description: Kubevirt Hyperconverged
    displayName: Kubevirt Hyperconverged
  defaultConfigs:
    - group: addon.open-cluster-management.io
      resource: addontemplates
      name: kubevirt-hyperconverged
  installStrategy:
    type: Placements
    placements:
    - name: openshift-cnv
      namespace: <installer_namespace>
----

> . Run `oc apply -f <filename.yaml>` to apply your resource.

Reviewer: ```suggestion
. Run the following command to apply your resource:

+
[source,bash]
----
oc apply -f <filename.yaml
----
```

link:https://issues.redhat.com/browse/ACM-14343[ACM-14343]

[#add-on-stuck-upgrade]
== The _kubevirt-hyperconverged_ add-on stuck in `in progress` status during upgrade from {acm-short} {product-version-prev} to {product-version}

During an upgrade from {acm-short} {product-version-prev} to {product-version}, the `kubevirt-hyperconverged-operator` resource is not cleaned up and you receive the following error in the `ManifestWork`:

[source,bash]
----
Failed to apply manifest: operatorpolicies.policy.open-cluster-management.io
"kubevirt-hyperconverged-operator" is forbidden: User
"system:serviceaccount:open-cluster-management-agent:klusterlet-work-sa"
cannot get resource "operatorpolicies" in API group
"policy.open-cluster-management.io" in the namespace "open-cluster-management-policies"
----

To resolve the issue, complete the following steps:

. Remove existing resources. Run the following commands:

+
[source,bash]
----
oc delete clustermanagementaddon kubevirt-hyperconverged-operator --ignore-not-found
oc delete addontemplate kubevirt-hyperconverged-operator --ignore-not-found
----

> . Delete previous `ClusterManagementAddOn` which is version `v1alpha1`. Run the following command.

Author: ```suggestion
. Delete the previous `ClusterManagementAddOn` resource, which is version `v1alpha1`. Run the following command:
```

dataimage.metal3.io/target-0-0 patched
----

link:https://issues.redhat.com/browse/ACM-21889[ACM-21889]

[#add-on-stuck-upgrade]

> == The _kubevirt-hyperconverged_ add-on failed to apply in upgrade from {acm-short} {product-version-prev} to {product-version}

Reviewer 2: addon is working but status is stuck in "in progress"

> During an upgrade from {acm-short} {product-version-prev} to {product-version}, you cannot apply the `kubevirt-hyperconverged-operator` add-on and you receive an error similar to the following error:

Reviewer 2: kubevirt-hyperconverged-operator is not cleaned up

Reviewer 2: error in ManifestWork

[source,bash]
----
Failed to apply manifest: operatorpolicies.policy.open-cluster-management.io
"kubevirt-hyperconverged-operator" is forbidden: User
"system:serviceaccount:open-cluster-management-agent:klusterlet-work-sa"
cannot get resource "operatorpolicies" in API group
"policy.open-cluster-management.io" in the namespace "open-cluster-management-policies"
----

To resolve the issue, complete the following steps:

. Remove existing resources. Run the following commands:

+
[source,bash]
----
oc delete clustermanagementaddon kubevirt-hyperconverged-operator --ignore-not-found
oc delete addontemplate kubevirt-hyperconverged-operator --ignore-not-found
----

> . Get the installer namespace from the `kubevirt-hyperconverged` label. Run the following command and note the output for use in the next step:

Reviewer 2: ```suggestion
. Get the namespace that is specified in the `installer.namespace` label of the `kubevirt-hyperconverged` `ClusterManagementAddOn` resource. Run the following command and record the output for use in the next step:
```

+
[source,bash]
----
oc get clustermanagementaddon kubevirt-hyperconverged -o jsonpath='{.metadata.labels.installer\.namespace}'
----
. Create the `ClusterManagementAddOn` resource. Replace `<installer_namespace>` with the value that you retrieved in the previous step:

+
[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1beta1
kind: ClusterManagementAddOn
metadata:
  name: kubevirt-hyperconverged
  annotations:
    addon.open-cluster-management.io/lifecycle: "addon-manager"
spec:
  addOnMeta:
    description: Kubevirt Hyperconverged
    displayName: Kubevirt Hyperconverged
  defaultConfigs:
    - group: addon.open-cluster-management.io
      resource: addontemplates
      name: kubevirt-hyperconverged
  installStrategy:
    type: Placements
    placements:
    - name: openshift-cnv
      namespace: <installer_namespace>
----

. Run `oc apply ClusterManagementAddOn` to apply your resource.

> . Reorder the `ManifestWork` resources so that the RBAC resources are defined before the `OperatorPolicy` to ensure the service account is authorized before the policy is processed. Run the following command:

Author: How do they reorder?

Reviewer 2: We can remove this because when ClustermanagementAddon modified manifestwork is reapplied

Author: line 95-102 step and command removed

+
[source,bash]
----
oc get managedcluster -l 'acm/cnv-operator-install=true' \
  -o jsonpath='{.items[*].metadata.name}'
----

. Create a local YAML file that is named `manifestwork.yaml` with the following content. Replace `<cluster_name>` with the name of your managed cluster in all instances:

+
[source,yaml]
----
apiVersion: work.open-cluster-management.io/v1
kind: ManifestWork
metadata:
  name: addon-kubevirt-hyperconverged-deploy-0
  namespace: <cluster_name>
spec:
  manifestConfigs:

>   - feedbackScrapeType: Poll

Reviewer 2: We can remove blew all

Reviewer 2: in my script, I just wanted to make sure the change so we don't need this

Author: removed everything after `- feedbackScrapeType: Poll` -- I was gonna ask you if we could eliminate some of that. TY

Reviewer 2: Creating the ClusterManagementAddOn automatically generates a new ManifestWork, so everything after line 96 can be removed.

Author: ok, apply step is removed.

Reviewer 2: we can remove "Create a local YAML file that is named `manifestwork.yaml` with the following content. Replace `<cluster_name>` with the name of your managed cluster in all instances:
"

+
[source,bash]
----
oc delete clustermanagementaddon kubevirt-hyperconverged-operator --ignore-not-found
oc delete addontemplate kubevirt-hyperconverged-operator --ignore-not-found
----

. Get the installer namespace from the `kubevirt-hyperconverged` label. Run the following command and note the output for use in the next step:

+
[source,bash]
----
oc get clustermanagementaddon kubevirt-hyperconverged -o jsonpath='{.metadata.labels.installer\.namespace}'
----
. Create the `ClusterManagementAddOn` resource. Replace `<installer_namespace>` with the value that you retrieved in the previous step:

+
[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1beta1
kind: ClusterManagementAddOn
metadata:
  name: kubevirt-hyperconverged
  annotations:
    addon.open-cluster-management.io/lifecycle: "addon-manager"
spec:
  addOnMeta:
    description: Kubevirt Hyperconverged
    displayName: Kubevirt Hyperconverged
  defaultConfigs:
    - group: addon.open-cluster-management.io
      resource: addontemplates
      name: kubevirt-hyperconverged
  installStrategy:
    type: Placements
    placements:
    - name: openshift-cnv
      namespace: <installer_namespace>
----

> . Run `oc apply ClusterManagementAddOn` to apply your resource.

Author: especially double check this area

Reviewer 2: We don't need this. because "create clusterManagementAddon" previous step cover this

. Reorder the `ManifestWork` resources so that the RBAC resources are defined before the `OperatorPolicy` to ensure the service account is authorized before the policy is processed. Run the following command:

+
[source,bash]
----
oc get managedcluster -l 'acm/cnv-operator-install=true' \
  -o jsonpath='{.items[*].metadata.name}'
----

> .. Create a local YAML file that is named `manifestwork.yaml` with the following content. Replace `<cluster_name>` with the name of your managed cluster in all instances:

Author: ```suggestion
. Create a local YAML file that is named `manifestwork.yaml` with the following content. Replace `<cluster_name>` with the name of your managed cluster in all instances:
```

[source,bash]
----
Failed to apply manifest: operatorpolicies.policy.open-cluster-management.io
"kubevirt-hyperconverged-operator" is forbidden: User
"system:serviceaccount:open-cluster-management-agent:klusterlet-work-sa"
cannot get resource "operatorpolicies" in API group
"policy.open-cluster-management.io" in the namespace "open-cluster-management-policies"
----

To resolve the issue, complete the following steps:

. Remove existing resources. Run the following commands:

+
[source,bash]
----
oc delete clustermanagementaddon kubevirt-hyperconverged-operator --ignore-not-found
oc delete addontemplate kubevirt-hyperconverged-operator --ignore-not-found
----

. Get the namespace that is specified in the `installer.namespace` label of the `kubevirt-hyperconverged` `ClusterManagementAddOn` resource. Run the following command and record the output for use in the next step:

+
[source,bash]
----
oc get clustermanagementaddon kubevirt-hyperconverged -o jsonpath='{.metadata.labels.installer\.namespace}'
----

. Create the `ClusterManagementAddOn` resource. Replace `<installer_namespace>` with the value that you retrieved in the previous step:

> +

Reviewer 2: ```suggestion
+ Delete previous ClusterManagementAddOn which is v1alpha1
---
oc delete clustermanagementaddon kubevirt-hyperconverged
---
```

Author: If we are asking them to run a command, we need to add that to the step and tell then why they are running it. So we need another step to delete but this is not before the YAML that we introduced already, so maybe after this?

Author: Then we need to introduce the YAML. Pretend there is a screen reader walking a nonsighted user through this. So we have to be very clear why we are giving a command and YAML.

Reviewer 2: Great!

Reviewer 2: "Delete previous ClusterManagementAddOn which is v1alpha1" this is the reason why we have to delete

Author: I think the placement of it is off--it's after create and before the YAML that they create--should this be on line 93?

Reviewer 2: before create yaml  is the best

[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1beta1
kind: ClusterManagementAddOn
metadata:
  name: kubevirt-hyperconverged
  annotations:
    addon.open-cluster-management.io/lifecycle: "addon-manager"
spec:
  addOnMeta:
    description: Kubevirt Hyperconverged
    displayName: Kubevirt Hyperconverged
  defaultConfigs:
    - group: addon.open-cluster-management.io
      resource: addontemplates
      name: kubevirt-hyperconverged
  installStrategy:
    type: Placements
    placements:
    - name: openshift-cnv
      namespace: <installer_namespace>
----

> . Run `oc apply -f <filename.yaml>` to apply your resource.

Author: we need to remove line 94 or no?

Author: Please see this question.

Reviewer 2: we must create the new ClusterManagementAddOn so leave as it is

Author: ok, I thought so but it looked like the previous remove comment was close to this. TY.

[source,bash]
----
Failed to apply manifest: operatorpolicies.policy.open-cluster-management.io
"kubevirt-hyperconverged-operator" is forbidden: User
"system:serviceaccount:open-cluster-management-agent:klusterlet-work-sa"
cannot get resource "operatorpolicies" in API group
"policy.open-cluster-management.io" in the namespace "open-cluster-management-policies"
----

To resolve the issue, complete the following steps:

. Remove existing resources. Run the following commands:

+
[source,bash]
----
oc delete clustermanagementaddon kubevirt-hyperconverged-operator --ignore-not-found
oc delete addontemplate kubevirt-hyperconverged-operator --ignore-not-found
----

. Get the namespace that is specified in the `installer.namespace` label of the `kubevirt-hyperconverged` `ClusterManagementAddOn` resource. Run the following command and record the output for use in the next step:

+
[source,bash]
----
oc get clustermanagementaddon kubevirt-hyperconverged -o jsonpath='{.metadata.labels.installer\.namespace}'
----

. Create the `ClusterManagementAddOn` resource. Replace `<installer_namespace>` with the value that you retrieved in the previous step:

> + Delete previous ClusterManagementAddOn which is v1alpha1

Author: we need to reorder this now.

Reviewer 2: yes you right Delete previous CMA first then create new one

Author: ok, we may not need to meet then--I was thinking I needed to show this live TY @yiraeChristineKim
